crates/goose/src/providers/groq.rs

```diff
@@ -54,7 +54,7 @@ impl GroqProvider {
         })
     }
 
-    async fn post(&self, payload: Value) -> anyhow::Result<Value, ProviderError> {
```

Comment:
how does it fix the Groq API key issue? `use anyhow::Result;` is imported at the top, so i think this is a syntactical change?